analyzers/src/SonarAnalyzer.Common/Helpers/ProjectConfigReader.cs

```
@@ -0,0 +1,76 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace SonarAnalyzer.Helpers
+{
+    /// <summary>
+    /// This class reads and encapsulates <see cref="ProjectConfig"/>, exposing only the configuration our analyzers need.
+    /// </summary>
+    internal class ProjectConfigReader
+    {
+        private readonly Lazy<ProjectConfig> projectConfig;
+
+        public string AnalysisConfigPath => ProjectConfigValue?.AnalysisConfigPath;
+        public string FilesToAnalyzePath => ProjectConfigValue?.FilesToAnalyzePath;
+        public string OutPath => ProjectConfigValue?.OutPath;
+        public string ProjectPath => ProjectConfigValue?.ProjectPath;
+        public ProjectType? ProjectType => ProjectConfigValue?.ProjectType;
+        public string TargetFramework => ProjectConfigValue?.TargetFramework;
+
+        private ProjectConfig ProjectConfigValue => projectConfig.Value;
+
+        public ProjectConfigReader(AnalyzerOptions options)
+        {
+            projectConfig = new Lazy<ProjectConfig>(() => ReadContent(options));
+        }
+
+        private static ProjectConfig ReadContent(AnalyzerOptions options)
+        {
+            var sonarProjectConfig = options.AdditionalFiles.FirstOrDefault(IsSonarProjectConfig);
+            if (sonarProjectConfig == null || !File.Exists(sonarProjectConfig.Path))
```

Just thinking out loud about `File.Exists`: "SonarProjectConfig.xml" was added as additional file but doesn't exist. That's a serious violation of configuration of the project.

I think we should fail hard in that specific case as well, for the same reason as the other comment in "catch" line.